diff_resolution.py

```diff
@@ -0,0 +1,83 @@
+                ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_a_lines[raw_diff[0][n-1] + 1])
+                ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_b_lines[raw_diff[1][n-1] + 1])
+            else:
+                next_vals = get_next_idx_match(raw_diff, n)
+                if (next_vals[0] - last_vals[0]) == (next_vals[1] - last_vals[1]):
+                    last_vals[0] += 1
+                    last_vals[1] += 1
+                    #print(f"{n}\t\t\t{last_vals[0]}\t\t\t{last_vals[1]}\t\t\t*DIFF***")
+                    ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_a_lines[last_vals[0]])
+                    ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_b_lines[last_vals[1]])
+
+                elif (next_vals[0] - last_vals[0]) > (next_vals[1] - last_vals[1]):
+                    if last_vals[1] < (next_vals[1] - 1):
+                        last_vals[0] += 1
+                        last_vals[1] += 1
+                        #print(f"{n}\t\t\t{last_vals[0]}\t\t\t{last_vals[1]}\t\t\t*DIFF**1")
+                        ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_a_lines[last_vals[0]])
+                        ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_b_lines[last_vals[1]])
+                    else:
+                        last_vals[0] += 1
+                        last_vals[1] += 1
+                        #print(f"{n}\t\t\t{last_vals[0]}\t\t\t-1\t\t\t*DIFF**2")
+                        ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_a_lines[last_vals[0]])
+                        ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.ADDED, "")
+
+                elif (next_vals[0] - last_vals[0]) < (next_vals[1] - last_vals[1]):
+                    if last_vals[0] < (next_vals[0] - 1):
+                        last_vals[0] += 1
+                        last_vals[1] += 1
+                        #print(f"{n}\t\t\t{last_vals[0]}\t\t\t{last_vals[1]}\t\t\t*DIFF**3")
+                        ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_a_lines[last_vals[0]])
+                        ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_b_lines[last_vals[1]])
+
+                    else:
+                        #print(f"{n}\t\t\t-1\t\t\t{last_vals[1] + 1}\t\t\t*DIFF**4")
+                        ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.SAME, "")
+                        ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.SAME, file_b_lines[last_vals[1]])
+                        last_vals[0] += 1
+                        last_vals[1] += 1
+                else:
+                    #print(f"{n}\t\t\t{raw_diff[0][n]}\t\t\t{raw_diff[1][n]}\t\t\t*DIFF***")
+                    ochangeSetA.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_a_lines[raw_diff[0][n]])
+                    ochangeSetB.addChange(n, pmEnums.CHANGEDENUM.CHANGED, file_b_lines[raw_diff[1][n]])
+
+    return pmEnums.RESULT.GOOD
```

Comment:
Maybe add a check for some sort of fail state, maybe, an else statement added to be sure if none of your logic is executed, indicating something went wrong, return ERROR so we no something escaped your logic before it was done. We will have to review that possibility together though.